src/Amalgam/Parser.cpp

```diff
@@ -1070,39 +1096,52 @@ EvaluableNode *Parser::GetNodeFromRelativeCodePath(EvaluableNode *path)
 		if(index_node == nullptr)
 			return nullptr;
 
-		//if it's an assoc, then treat the index as a string
-		if(result->GetMappedChildNodes().size() > 0)
+		if(index_node->IsOrderedArray())
 		{
-			StringInternPool::StringID index_sid = EvaluableNode::ToStringIDIfExists(index_node);
-			EvaluableNode **found = result->GetMappedChildNode(index_sid);
-			if(found != nullptr)
-				return *found;
-			return nullptr;
+			//travers the nodes over each index to find the location
+			auto &index_ocn = index_node->GetOrderedChildNodesReference();
+			for(auto &index_node_element : index_ocn)
+			{
+				result = GetNodeRelativeToIndex(result, index_node_element);
+				if(result == nullptr)
+					break;
+			}
+
+			return result;
+		}
+		else //immediate
+		{
+			return GetNodeRelativeToIndex(result, index_node);
 		}
 
-		//otherwise treat the index as a number for a list
-		size_t index = static_cast<size_t>(EvaluableNode::ToNumber(index_node));
-		if(result->GetOrderedChildNodes().size() > index)
-			return result->GetOrderedChildNodes()[index];
+		return nullptr;
 	}
 
 	case ENT_TARGET:
 	{
 		//first parameter is the number of steps to crawl up in the parent tree
-		size_t steps_up = 1;
+		size_t target_steps_up = 1;
 		if(path->GetOrderedChildNodes().size() > 0)
-			steps_up = static_cast<size_t>(EvaluableNode::ToNumber(path->GetOrderedChildNodes()[0]));
+		{
+			double step_value = EvaluableNode::ToNumber(path->GetOrderedChildNodes()[0]);
+			if(step_value >= 1)
+				target_steps_up = static_cast<size_t>(step_value);
+			else
+				return nullptr;
```

Comment:
In a similar spot you had a branch that set the step value to 0 if it was (null), should this spot do that as well?